backend/services/implementations/testService.ts

```diff
@@ -272,6 +277,8 @@ class TestService implements ITestService {
       test.isNew = true;
       test.status = AssessmentStatus.DRAFT;
       test.save();
+
+      await this.incrementImageCounts(test.questions);
```

Comment:
I think it might be better to invert the order of these two lines. If `save` passes but `increment` fails, then we might end up deleting an image before it's actually unused which is bad UX. Otherwise, we might end up retaining images for longer than we need them, but I think this is more reasonable as we can always manually delete mismatched images.

Also, shouldn't we be `await`ing on `test.save()`? I think right now the endpoint returns before the test is actually registered in the database. nbd before, but now that the logic order matters I think we ought to await